### PR TITLE
Add theseus.geometry.functional.so3.inverse()

### PR DESCRIPTION
**Pull request by fantaosha**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by luisenp — APPROVED** [review]
LGTM.

**Comment by luisenp** [comment]
Can we also close #366?

File: tests/geometry/functional/test_so3.py [hunk 1]
```diff
@@ -28,3 +28,13 @@ def test_adjoint(batch_size: int, dtype: torch.dtype):
...
     check_lie_group_function(so3, "adjoint", TEST_EPS, group)
+
+
+@pytest.mark.parametrize("batch_size", [1, 20, 100])
+@pytest.mark.parametrize("dtype", [torch.float32, torch.float64])
+def test_inverse(batch_size: int, dtype: torch.dtype):
+    rng = torch.Generator()
+    rng.manual_seed(0)
+    tangent_vector = torch.rand(batch_size, 3, dtype=dtype, generator=rng)
+    group = so3.exp(tangent_vector)
```
Comment (mhmukadam) [inline tests/geometry/functional/test_so3.py:39]: would we rather generate a rand so3 or does that function also internally call exp?

**Review by mhmukadam — APPROVED** [review]

File: tests/geometry/functional/test_so3.py (hunk 1 above)
Comment (fantaosha) [reply tests/geometry/functional/test_so3.py:39]: I'm working on this now. Should be pushed in the next two PRs.

**Pull request merged**